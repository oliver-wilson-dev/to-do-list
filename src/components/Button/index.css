@import "../../styles/variables.css";

.button {
    padding: 10px 40px;
    border: 3px solid #f0134d;
    box-sizing: border-box;
    text-transform:uppercase;
    font-weight:400;
    text-align:center;
    transition: all 0.15s;
    font-size: 18px;
    background-color: var(--primaryBackgroundColour);
}
