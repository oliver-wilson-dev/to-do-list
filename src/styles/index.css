/* This is where any shared styles should live */
@import "./variables.css";

body {
  margin: 0;
  padding: 0;
  background-color: var(--primaryBackgroundColour);
}

h1 {
  font-family: "futura-pt", sans-serif;
}

h2,
h3 {
  font-family: "futura-pt", sans-serif;
  font-weight: 900;
  letter-spacing: 1px;
}

p {
  line-height: 2em;
  font-size: 16px;
}

a {
  text-decoration: none;
  font-family: "futura-pt", sans-serif;
  color: rgb(21, 64, 204);
}