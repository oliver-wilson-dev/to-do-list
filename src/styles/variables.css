:root {
    --primaryBackgroundColour:  #f8f8f8;
    --primaryBorderColour: #0b0a0a;
    --primaryTextColor: #333;
}