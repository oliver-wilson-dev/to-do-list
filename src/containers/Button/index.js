import { connect } from 'react-redux';
import Button from '../../components/Button';
import incrementCount from '../../state/actions/incrementCount';

const mapDispatchToProps = dispatch => ({
  incrementCount: () => dispatch(incrementCount())
});

export default connect(undefined, mapDispatchToProps)(Button);
