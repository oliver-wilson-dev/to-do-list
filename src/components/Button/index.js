import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const Button = ({ incrementCount }) => (
  <button className={styles.button} onClick={incrementCount} type="button">Click me!</button>
);

Button.propTypes = {
  incrementCount: PropTypes.func.isRequired
};

export default Button;
