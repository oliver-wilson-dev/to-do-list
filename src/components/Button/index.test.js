import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

import styles from './index.css';

const mockOnClick = jest.fn();

const defaultProps = { onClick: mockOnClick };

const render = overrideProps => shallow(<Button {...defaultProps} {...overrideProps} />);

describe('<Button/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when there are children to be rendered', () => {
    it('renders correctly', () => {
      expect(render({ children: <span>hello world</span> })).toMatchSnapshot();
    });
  });

  describe('when there are additionalStyles to be applied', () => {
    it('renders correctly', () => {
      const mockAdditionalStyles = 'test-additional-styles';
      expect(render({ additionalStyles: mockAdditionalStyles })
        .find('button')
        .prop('className'))
        .toEqual(`${styles.button} ${mockAdditionalStyles}`);
    });
  });

  describe('when the button is clicked', () => {
    it('should call the onClick handler', () => {
      const mockOnClick = jest.fn();
      const wrapper = render({ onClick: mockOnClick });

      wrapper.find('button').simulate('click');

      expect(mockOnClick).toHaveBeenCalled();
    });
  });
});
