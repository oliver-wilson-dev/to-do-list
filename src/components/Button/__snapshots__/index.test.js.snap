// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> component should render correctly 1`] = `
<button
  className="button "
  onClick={[MockFunction]}
  type="button"
/>
`;

exports[`<Button/> component when there are children to be rendered renders correctly 1`] = `
<button
  className="button "
  onClick={[MockFunction]}
  type="button"
>
  <span>
    hello world
  </span>
</button>
`;
